File: java/pe/src/ktn/pe/Pe5.java
package ktn.pe;

import java.util.List;
import java.util.ArrayList;

public class Pe5 implements Pe {
    private int to = 20;

    public static long pe5(int to) {
        long mul = 1L;

        if (to < 2) {
            return mul;
        }
        mul = 2L;
        List<Integer> factors = new ArrayList<Integer>();
        factors.add(2);

        int x;
        for (int i = 3; i < to + 1; ++i) {
            x = i;
            for (int factor : factors) {
                if ((x % factor) == 0) {
                    x /= factor;
                }
                if (x < factor) {
                    break;
                }
            }

            if (x > 1) {
                factors.add(x);
                mul *= x;
            }
        }

        return mul;
    }

    @Override
    public void setArgs(String[] args) {
        if (args != null && args.length > 0) {
            try {
                to = Integer.parseInt(args[0]);
            } catch (NumberFormatException e) {
                //
            }
        }
    }

    @Override
    public void solve() {
        System.out.println(pe5(to));
    }

}

File: kt/pe/src/main/kotlin/ktn/pe/Pe5.kt
package ktn.pe

class Pe5 : Pe {
    private var to = 20

    override val problemNumber: Int
        get() = 5

    fun pe5(to: Int): Long {
        var mul = 1L

        if (to < 2) {
            return mul
        }
        mul = 2L
        val factors = mutableListOf<Int>()
        factors.add(2)

        var x: Int
        for (i in 3 until to + 1) {
            x = i
            for (factor in factors) {
                if (x % factor == 0) {
                    x /= factor
                }
                if (x < factor) {
                    break
                }
            }

            if (x > 1) {
                factors.add(x)
                mul *= x.toLong()
            }
        }

        return mul
    }

    override fun setArgs(args: Array<String>) {
        if (args.isNotEmpty()) {
            to = args[0].toInt(10)
        }
    }

    override fun solve() {
        System.out.println(PeUtils.format(this, pe5(to)))
    }

    override fun run() {
        solve()
    }

}
